test_conventions.TestEncodeCFVariable failing on master for Appveyor Python 2.7 build

I have on idea what's going on here but maybe somebody who knows Windows better has a guess:
```
================================== FAILURES ===================================
_________________ TestEncodeCFVariable.test_missing_fillvalue _________________
self = <xarray.test.test_conventions.TestEncodeCFVariable testMethod=test_missing_fillvalue>
    def test_missing_fillvalue(self):
        v = Variable(['x'], np.array([np.nan, 1, 2, 3]))
        v.encoding = {'dtype': 'int16'}
        with self.assertWarns('floating point data as an integer'):
>           conventions.encode_cf_variable(v)
xarray\test\test_conventions.py:523: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Python27-conda32\lib\contextlib.py:24: in __exit__
    self.gen.next()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <xarray.test.test_conventions.TestEncodeCFVariable testMethod=test_missing_fillvalue>
message = 'floating point data as an integer'
    @contextmanager
    def assertWarns(self, message):
        with warnings.catch_warnings(record=True) as w:
            warnings.filterwarnings('always', message)
            yield
            assert len(w) > 0
>           assert all(message in str(wi.message) for wi in w)
E           AssertionError: NameError: all(<generator object <genexpr> at 0x0617D170>)  << global name 'message' is not defined
xarray\test\__init__.py:140: AssertionError
============== 1 failed, 970 passed, 67 skipped in 70.58 seconds ==============
```

I could understand a warning failing to be raised, but the `NameError` is especially strange.

We're getting a sensible `AssertionError` now:
```
_________________ TestEncodeCFVariable.test_missing_fillvalue _________________
self = <xarray.test.test_conventions.TestEncodeCFVariable testMethod=test_missing_fillvalue>
    def test_missing_fillvalue(self):
        v = Variable(['x'], np.array([np.nan, 1, 2, 3]))
        v.encoding = {'dtype': 'int16'}
        with self.assertWarns('floating point data as an integer'):
>           conventions.encode_cf_variable(v)
xarray\test\test_conventions.py:523: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Python27-conda32\lib\contextlib.py:24: in __exit__
    self.gen.next()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <xarray.test.test_conventions.TestEncodeCFVariable testMethod=test_missing_fillvalue>
message = 'floating point data as an integer'
    @contextmanager
    def assertWarns(self, message):
        with warnings.catch_warnings(record=True) as w:
            warnings.filterwarnings('always', message)
            yield
            assert len(w) > 0
>           assert all(message in str(wi.message) for wi in w)
E           AssertionError
xarray\test\__init__.py:140: AssertionError
```

I guess this was probably a py.test bug.

--- a/xarray/test/__init__.py
+++ b/xarray/test/__init__.py
@@ -137,7 +137,7 @@ class TestCase(unittest.TestCase):
             warnings.filterwarnings('always', message)
             yield
             assert len(w) > 0
-            assert all(message in str(wi.message) for wi in w)
+            assert any(message in str(wi.message) for wi in w)
 
     def assertVariableEqual(self, v1, v2):
         assert as_variable(v1).equals(v2), (v1, v2)

